### PR TITLE
[xh]Replace openai function call with AI client function 

### PR DESCRIPTION
**Pull request by matrixstone**
# Description
In wizard class, there is openai function call to classify "block_type", "pipeline_type" based on code description.

In this PR, I replaces this openai function with dedicated "find_block_params" in AI client funciton.
So each AI client can implement it. So wizard class becomes fully AI client isolated.

# How Has This Been Tested?
Tested with 10 different code description to compare open ai classification response with hugging face classification response: https://docs.google.com/spreadsheets/d/1-fongvJv0lrtNp06wbnDqxmPNp66lCBJ-8TVKuftDds/edit?usp=sharing

### Tested with Mage UI
Generate block with code description:
2_loader_test
2_transformer
2_exporter

Generate pipeline with description:
2_generate_pipeline
2_pipeline_generation_result

# Checklist
- [X] The PR is tagged with proper labels (bug, enhancement, feature, documentation)
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas

cc:
@wangxiaoyou1993

### CODE REVIEW

**Comment by wangxiaoyou1993** [comment]
You can rebase master branch to fix the CI failure

**Comment by matrixstone** [comment]
👍👍Fixed the CI! Thank you @wangxiaoyou1993

File: mage_ai/ai/ai_client.py [hunk 1]
```diff
@@ -24,3 +24,22 @@ async def inference_with_prompt(
...
+            block_description: str
+    ):
+        """
+        Based on the block descriptionn, find the block type,
+        block language, pipeline type, and config for this block.
+
+        Args:
+          block_description: description of the block functions.
+
+        Returns:
```
Comment (wangxiaoyou1993) [inline mage_ai/ai/ai_client.py:39]: I would recommend returning a dictionary with these key values, which will make it easy to extend.

File: mage_ai/ai/hugging_face_client.py [hunk 2]
```diff
@@ -27,6 +61,60 @@ def __init__(self, hf_config: HuggingFaceConfig):
             os.getenv('HUGGINGFACE_API')
         print(f'Using Hugging Face API: {self.api}')
 
+    def __parse_function_args(self, function_args: json):
```
Comment (wangxiaoyou1993) [inline mage_ai/ai/hugging_face_client.py:64]: json is not a type. i guess you want to use `Dict` type

**Review by wangxiaoyou1993 — COMMENTED** [review]
Can you also test with Mage UI?

File: mage_ai/ai/ai_client.py (hunk 1 above)
Comment (matrixstone) [reply mage_ai/ai/ai_client.py:39]: Make total sense. Updated it to return dictionary instead.

**Comment by matrixstone** [comment]
@wangxiaoyou1993 Updated the PR following comments and also tested with Mage UI.

The hugging face client functionally works by generating blocks and pipeline. But I look into the actual code it generated, the accuracy is something we can improve in the next step.

**Review by wangxiaoyou1993 — APPROVED** [review]

**Pull request merged**